web/modules/custom/weather/src/Services/WeatherApi.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Drupal\weather\Services;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use GuzzleHttp\ClientInterface;
+
+/**
+ * Class WeatherApi.
+ */
+class WeatherApi {
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    protected ConfigFactoryInterface $configFactory,
+    protected ClientInterface $httpClient,
+  ) {}
+
+  /**
+   * Weather Api Validation.
+   */
+  public function validateApi(string $apiKey): bool {
+    try {
+      $client = $this->httpClient->get('http_client');
+      $client->get('https://api.openweathermap.org/data/2.5/weather?q=London&appid=' . $apiKey);
+    }
+    catch (\Exception $error) {
+      return FALSE;
+    }
+    return TRUE;
+  }
+
+  /**
+   * Weather City Validation.
+   */
+  public function validateCityApi(string $cityName): bool {
+    $config = $this->configFactory->get('weather.settings');
+    $apiKey = $config->get('api_admin_key');
+    try {
+      $client = \Drupal::httpClient();
```

Comment:
Forgot about Dependency Injection :)
It must be ->
$client = $this->httpClient->get('http_client');